docker_mcp/services/host.py

```
@@ -719,204 +764,567 @@
         for priority_path in priorities:
             for path in paths:
                 if priority_path in path:
                     return path
-        
+
         # Return first available path
         return paths[0] if paths else None
 
-    async def _test_ssh_connection(self, hostname: str, user: str, port: int = 22, identity_file: str | None = None) -> bool:
-        """Test SSH connection with raw parameters before adding host to config.
+    async def handle_action(self, action, **params) -> dict[str, Any]:
```

Ruff (0.12.2)

`handle_action` is too complex (37 > 10)

(C901)